staging/kos/pkg/controllers/connectionagent/connectionagent.go

```diff
@@ -0,0 +1,1091 @@
+		err = ca.processExistingAtt(att)
+	} else if deleted {
+		// If we are here the attachment has been deleted
+		err = ca.processDeletedAtt(attNSN)
+	}
+	return err
+}
+
+// getAttachment attempts to determine the univocal version of the NetworkAttachment with namespaced name attNSN. If it succeeds it
+// returns the attachment if it is found in an Informer cache or attDeleted set to true if it could not be found in any cache (e.g.
+// because it has been deleted). If the current attachment version cannot be determined without ambiguity, an error is returned.
+// An attachment is considered amibguous if it either has been seen with more than one vni in a remote attachments cache, or if
+// it is found both in the local attachments cache and a remote attachments cache.
+func (ca *ConnectionAgent) getAttachment(attNSN k8stypes.NamespacedName) (att *netv1a1.NetworkAttachment, attDeleted bool, err error) {
+	// Retrieve the VNIs where the attachment could be as a remote attachment
+	attSeenInVNIs := ca.getAttVNIs(attNSN)
+	nbrOfVNIs := len(attSeenInVNIs)
+	if nbrOfVNIs > 1 {
+		// If the attachment could be a remote one in more than one VNI, return an error. This will cause
+		// the att reference to be requeued, and hopefully by the time it is processed again the
+		// inconsistency as to which is the vni of the attachment has been resolved
+		err = fmt.Errorf("Attachment %s has inconsistent state, found in the following VNIs: %#v", attNSN, attSeenInVNIs)
+		return
+	}
+
+	// If the attachment has been seen in exactly one VNI lookup it up in
+	// the remote attachments cache for the VNI with which it's been seen
+	var (
+		vni                  uint32
+		attAsRemote          *netv1a1.NetworkAttachment
+		remAttCacheLookupErr error
+	)
+	if nbrOfVNIs == 1 {
+		vni := attSeenInVNIs[0]
+		remoteAttsLister := ca.getRemoteAttListerForVNI(vni)
+		if remoteAttsLister != nil {
+			attAsRemote, remAttCacheLookupErr = remoteAttsLister.Get(attNSN.Name)
+		}
+	}
+
+	// Lookup the attachment in the local attachments cache
+	attAsLocal, localAttCacheLookupErr := ca.localAttsLister.NetworkAttachments(attNSN.Namespace).Get(attNSN.Name)
+
+	switch {
+	case (remAttCacheLookupErr != nil && !k8serrors.IsNotFound(remAttCacheLookupErr)) ||
+		(localAttCacheLookupErr != nil && !k8serrors.IsNotFound(localAttCacheLookupErr)):
+		// If we're here at least one of the two lookups failed. This should never happen. No point in retrying.
+		glog.V(1).Infof("attempt to retrieve attachment %s with lister failed: %s. This should never happen, hence a reference to %s will not be requeued",
+			attNSN,
+			aggregateErrors("\n\t", remAttCacheLookupErr, localAttCacheLookupErr).Error(),
+			attNSN)
+		err = nil
+	case attAsLocal != nil && attAsRemote != nil:
+		// If we're here the attachment has been found in both caches: return an error. This will cause its reference
+		// to be requeued, and hopefully by the time it's processed again the inconsistency has been resolved
```

Comment:
We do not need to requeue in this case; the deletion that removes this condition will enqueue.

Reply:
Fixed in 5aca980.